compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ImageComposeScene.skikoMain.kt

```diff
@@ -107,13 +108,17 @@ class ImageComposeScene(
     height: Int,
     density: Density = Density(1f),
     coroutineContext: CoroutineContext = Dispatchers.Unconfined,
+    effectDispatcher: CoroutineDispatcher? = null,
+    recomposeDispatcher: CoroutineDispatcher? = null,
```

Comment:
1. I would like to avoid any ambiguous arguments in a public constructor. So far, we only have one argument `coroutineContext`. Now we have 3 arguments, which purpose isn't well defined from external point of view. To be honest, initially there was also no well defined purpose of `coroutineContext` too.

2. Having `null` as a default argument isn't very flexible, because it usually means that we have some internal functionality that are not accessible outside. Here we have `FlushCoroutineDispatcher`, and once we override the dispatcher, we lose functionality for which `FlushCoroutineDispatcher` was written. For example, if we pass `recomposeDispatcher = mySingleThreadDispatcher` we will have a different behavior of the `render` function - it will not perform all pended recompositions before rendering the frame.

There is no easy and well defined way to fix these issues, we need to investigate. 

If we only need to fix a bug - ability to control delay's, we can try to find another way, without changing public API. There were some ways that came to my mind:
1. remove `FlushCoroutineDispatcher` and add some method to the common `Recomposer`. Something like `performRecomposition`. Isn't easy, because we need to think about public API of `Recomposer` and discuss it with Google.
2. Support delays in `FlushCoroutineDispatcher` (extending it from Delay). Not sure if it is a good option, because `Delay` is `InternalCoroutinesApi` (but we done that for `org.jetbrains.skiko.MainUIDispatcher`)
3. Try to control recompositions with the current functionality of Recomposer and without using a custom dispatcher.
4. Or maybe there is no issue in the first place, and we can just remove `FlushCoroutineDispatcher`.

I can look at it soon (as it was written by me), and decide what we can do.

Reply:
It's not only about recomposition. We have double click and long click, which rely on withTimeoutOrNull (uses delay internally).

In my opinion, it would be good if be able to control all coroutines dispatching in tests.

Reply:
> We have double click and long click, which rely on withTimeoutOrNull 

Indeed. Found the old comment why it is used for effects:
```
    // We use FlushCoroutineDispatcher for effectDispatcher not because we need `flush` for
    // LaunchEffect tasks, but because we need to know if it is idle (hasn't scheduled tasks)
    private val effectDispatcher = FlushCoroutineDispatcher(coroutineScope)
```

Reply:
I removed this changed in ComposeScene from this PR (to let it move forward) and reworked the tests to use new experimental `fun runSkikoComposeUiTest`. Please have a look